parsl/app/python.py

```diff
@@ -45,6 +45,29 @@ def __call__(self, *args, **kwargs):
         else:
             dfk = self.data_flow_kernel
 
+        walltime = self.kwargs.get('walltime')
+        if walltime is not None:
+            def timeout(f, walltime):
+                def wrapper(*args, **kwargs):
+                    import threading
+
+                    def stop(thread):
+                        import ctypes
+                        import parsl.app.errors
```

Comment:
i have this niggling doubt that doing these import here means that the `stop` routine might take a while. it's probably desirable for the stop routine to get the exception set as fast as it can, as I think there's a race between i) the Timer expiring and the exception being set; and ii) the function successfully completing, so that the async exception gets thrown into some later part of the execution. Making `stop` be as fast as possible mitigates that.

Reply:
Great point-- agreed. Thanks for catching that.